pkg/cmd/roachtest/roachtestutil/disk_usage.go

```diff
@@ -128,39 +128,23 @@ func NewDiskUsageTracker(
 	return &DiskUsageTracker{c: c, l: diskLogger}, nil
 }
 
-// GetDiskUsageInBytes does what's on the tin. nodeIdx starts at one.
+// GetDiskUsageInBytes executes the command `du {store-dir}` on node `nodeIdx`.
+// Optional args are appended, so it can parse the result to bytes. nodeIdx starts at one.
 func GetDiskUsageInBytes(
 	ctx context.Context, c cluster.Cluster, logger *logger.Logger, nodeIdx int,
 ) (int, error) {
 	var result install.RunResultDetails
-	for {
-		var err error
-		// `du` can warn if files get removed out from under it (which
-		// happens during RocksDB compactions, for example). Discard its
-		// stderr to avoid breaking Atoi later.
-		// TODO(bdarnell): Refactor this stack to not combine stdout and
-		// stderr so we don't need to do this (and the Warning check
-		// below).
-		result, err = c.RunWithDetailsSingleNode(
-			ctx,
-			logger,
-			option.WithNodes(c.Node(nodeIdx)),
-			"du -sk {store-dir} 2>/dev/null | grep -oE '^[0-9]+'",
-		)
-		if err != nil {
-			if ctx.Err() != nil {
-				return 0, ctx.Err()
-			}
-			// If `du` fails, retry.
-			// TODO(bdarnell): is this worth doing? It was originally added
-			// because of the "files removed out from under it" problem, but
-			// that doesn't result in a command failure, just a stderr
-			// message.
-			logger.Printf("retrying disk usage computation after spurious error: %s", err)
-			continue
-		}
-
-		break
+	var err error
+	// Retry a few times since `du` can warn if files get removed out from under
+	// it (which happens during pebble compactions, for example).
+	result, err = c.RunWithDetailsSingleNode(
+		ctx,
+		logger,
+		option.WithNodes(c.Node(nodeIdx)).WithShouldRetryFn(install.AlwaysTrue),
```

Comment:
Realized I was completely misusing the `retry` package, and it wasn't actually retrying. Switched to the roachprod retry instead :) 

Want to double check my retry reasoning here though. We want to retry here since we are parsing the result and a stray warning can throw that off. We don't need to retry in `saveDiskUsageToLogsDir` since we are just writing to logs and the warnings are benign.

Reply:
> We want to retry here since we are parsing the result and a stray warning can throw that off. We don't need to retry in saveDiskUsageToLogsDir since we are just writing to logs and the warnings are benign.

Exactly! If `grep` fails to match a number, the entire command exits with an error, which would trigger a retry attempt.